clients/rocsolvercommon/rocsolver_test.hpp

```diff
@@ -53,17 +53,17 @@ inline void rocsolver_bench_output(T arg, Ts... args)
     rocsolver_bench_output(args...);
 }
 
-// template <typename T, std::enable_if_t<!is_complex<T>, int> = 0>
-// inline T sconj(T scalar)
-// {
-//     return scalar;
-// }
+template <typename T, std::enable_if_t<!is_complex<T>, int> = 0>
+inline T sconj(T scalar)
+{
+    return scalar;
+}
 
-// template <typename T, std::enable_if_t<is_complex<T>, int> = 0>
-// inline T sconj(T scalar)
-// {
-//     return std::conj(scalar);
-// }
+template <typename T, std::enable_if_t<is_complex<T>, int> = 0>
+inline T sconj(T scalar)
+{
+    return std::conj(scalar);
+}
```

Comment:
IIRC, Juan originally added sconj as a workaround for some compiler or library issue in ROCm 3.9. If that issue has been resolved, perhaps we can just use `std::conj`?

Reply:
Switching it over seems to work fine. We'll probably want to make the same change to rocSOLVER.

Reply:
It works because we are injecting `conj` for real cases in the std namespace here `clients/include/complex.hpp`
rocBLAS does not do that for rocblas_complex types, so this won't work in rocSOLVER.
For some reason, std::conj does not like rocblas_complex types...

Reply:
Oh, I see what's going on now. The actual [`std::conj`](https://en.cppreference.com/w/cpp/numeric/complex/conj) has a specialization that takes a real input (by value), but it returns `std::complex`. We have defined an overload of `std::complex` that takes a real input (by reference) and returns a real output. That's very confusing.

The [section of clients/include/complex.hpp which does that](https://github.com/ROCmSoftwarePlatform/hipSOLVER/blob/develop/clients/include/complex.hpp#L156-L207) is not good. There's very few cases in which it's valid to insert something into the `std::` namespace, and I would not be surprised if that code is causing half the problems that `sconj` is solving.

My suggestion: when we swap out `hipsolverComplex` for `hipFloatComplex`, we should drop all those overloads from the `std` namespace. We can define our own functions on an as-needed basis.